docs/manual/cluster-admin/how-to-customize-alerts.md

```diff
@@ -0,0 +1,155 @@
+# How to Customize Alerts
+
+OpenPAI supports the customization of alert rules and corresponding handling actions.
+The alert rules are managed by `prometheus` service and the handling actions are managed by `alert-manager` service.
+
+In this document, we will introduce existing alerts & actions, their matching methods, and how to add new customized alerts & actions.
+
+## Existing Alerts/Actions & How to Match Them 
+
+### Existing Alerts
+
+OpenPAI uses `Prometheus` to monitor system metrics.
+We provide various alerts by defining rules on virtual_clusters, GPU utilization, etc.
+If OpenPAI is deployed, you can then visit `your_master_ip/prometheus/alerts` to see the details of alerts, including their definition and status.
+
+For example, `PAIJobGpuPercentLowerThan0_3For1h` alert rule is by default configured as :
+
+``` yaml
+alert: PAIJobGpuPercentLowerThan0_3For1h
+expr: avg(task_gpu_percent{virtual_cluster=~"default"}) BY (job_name) < 0.3
+for: 1h
+```
+
+It means that the job on virtual cluster `default` with task level average GPU percent lower than `30%` for more than `1 hour` will fire the alert `PAIJobGpuPercentLowerThan0_3For1h`.
+
+### Existing Actions
+
+OpenPAI uses `alert-manager` service for alert handling. We provided three types of actions: 
+
+* webportal-notification: Show alerts on the home page of webportal.
+* email-notification: Send email to the assigned receiver.
+* webhook-actions: Send alert via a POST request and then adapt the request to customized actions. We now support these actions:
+    - stop-job: Stop jobs by adapting the POST request to the corresponding OpenPAI REST API.
+
+`webportal-notification` is always enabled, all the alerts will be shown on the webportal.
+
+To enable `email-notification` and `webhook-actions`, administrators need to configure necessary information in the `alert-manager` field of `service-configuration.yml`: 
+
+```yaml
+alert-manager:
+  port: 9093 # optional, do not change this if you do not want to change the port alert-manager is listening on
+  email_configs: # email-notification will only be enabled when this field is not empty
+    receiver: your_addr@example.com
+    smtp_url: smtp.office365.com:587
+    smtp_from: alert_sender@example.com
+    smtp_auth_username: alert_sender@example.com
+    smtp_auth_password: password_for_alert_sender
+  webhook_configs: # webhook-actions will only be enabled when  this field is not empty
+    port: 9095 # optional, do not change this if you do not want to change the port alert-handler is listening on
+    pai_bearer_token: 'your_application_token_for_rest_server'
```

Comment:
Since we only have stop-job action. Currently, it's ok.
But I hope we can config like this:
```yaml
webhook_configs:
  port: 9095
  pai_bearer_token: 'your_application_token_for_rest_server'
  actions:
     - stop_job
     - cordon_node
```
Admin can change the config to apply different actions.

So the workflow can be this:
1. We provide many predefined alert rules and actions.
2. If admin want to apply this alert & action, let he put the alert rules under Prometheus, and change this config to enable action